File: cpp/001-010/Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x == 0) return true;
        if (x < 0) return false;
        if (x % 10 == 0) return false;
        
        int sum = 0;
        while (x > sum) {
            sum = sum * 10 + x % 10;
            x /= 10;
        }
        return (x == sum) || (x == sum / 10);
    }
    
};

File: scala/0001-0010/Palindrome Number.scala
object Solution {
    def isPalindrome(x: Int): Boolean = {
        if (x < 0) return false
        val s = x.toString
        
        for( i <- 0 to s.length()/2 ) {
            if (s(i) != s(s.length() - i - 1)) return false
        }
        
        true
    }
}
